doc/index.rst

```diff
@@ -68,6 +68,7 @@ Catalyst
    :caption: Development
    :hidden:
 
+   dev/debugging
    dev/architecture
    modules/frontend
    modules/mlir
```

Comment:
@josh146 Which section do you think this should go into? (and in what order)

![image](https://github.com/PennyLaneAI/catalyst/assets/29467667/ed273090-9b07-4e01-9c0a-460a481912b3)

Reply:
@dime10 I would put it at the bottom, just above Roadmap :)